modules/carto/src/api/endpoints.ts

```diff
@@ -11,28 +11,58 @@ function buildUrlFromBase(apiBaseUrl: string, endpoint: V3Endpoint) {
   return `${apiBaseUrl}${suffix}`;
 }
 
-export function buildMapsUrlFromBase(apiBaseUrl: string): string {
-  return buildUrlFromBase(apiBaseUrl, 'maps');
+export function buildPublicMapUrl({
+  apiBaseUrl,
+  cartoMapId
+}: {
+  apiBaseUrl: string;
+  cartoMapId: string;
+}): string {
+  return `${buildUrlFromBase(apiBaseUrl, 'maps')}/public/${cartoMapId}`;
```

Comment:
Optional: Maybe the name of this function, or structure of this query building, could be clarified a little? In my mind a "URL" usually has any of {protocol, domain, path, query, fragment}, and so it isn't necessarily safe to append new paths at the end of a URL. Or in this case, I couldn't tell it was safe until reading the function implementation.

To me any of these would be easier to read...

```js
// (a)
return joinPath(apiBaseUrl, 'v3', 'maps', 'public', cartoMapId);

// (b)
return joinPath(apiBaseUrl, `/v3/maps/public/${cartoMapId}`);
```

... where `joinPath` takes care of missing or redundant `/` separators.